x-pack/platform/plugins/shared/alerting/server/maintenance_window_events/task.ts

```diff
@@ -0,0 +1,296 @@
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import type { KueryNode } from '@kbn/es-query';
+import { nodeBuilder, nodeTypes } from '@kbn/es-query';
+import { MAINTENANCE_WINDOW_SAVED_OBJECT_TYPE } from '../../common';
+import type { AlertingPluginsStart } from '../plugin';
+import type { MaintenanceWindowAttributes } from '../data/maintenance_window/types';
+import {
+  transformMaintenanceWindowAttributesToMaintenanceWindow,
+  transformMaintenanceWindowToMaintenanceWindowAttributes,
+} from '../application/maintenance_window/transforms';
+import { generateMaintenanceWindowEvents } from '../application/maintenance_window/lib/generate_maintenance_window_events';
+import { stateSchemaByVersion, emptyState, type LatestTaskStateSchema } from './task_state';
+
+export const MAINTENANCE_WINDOW_EVENTS_TASK_TYPE = 'maintenance-window:generate-events';
+
+export const MAINTENANCE_WINDOW_EVENTS_TASK_ID = `${MAINTENANCE_WINDOW_EVENTS_TASK_TYPE}-generator`;
+export const SCHEDULE: IntervalSchedule = { interval: '1d' };
+
+export function initializeMaintenanceWindowEventsGenerator(
+  logger: Logger,
+  taskManager: TaskManagerSetupContract,
+  coreStartServices: Promise<[CoreStart, AlertingPluginsStart, unknown]>
+) {
+  registerMaintenanceWindowEventsGeneratorTask(logger, taskManager, coreStartServices);
+}
+
+export async function scheduleMaintenanceWindowEventsGenerator(
+  logger: Logger,
+  taskManager: TaskManagerStartContract
+) {
+  try {
+    await taskManager.ensureScheduled({
+      id: MAINTENANCE_WINDOW_EVENTS_TASK_ID,
+      taskType: MAINTENANCE_WINDOW_EVENTS_TASK_TYPE,
+      schedule: SCHEDULE,
+      state: emptyState,
+      params: {},
+    });
+  } catch (e) {
+    logger.error(`Error scheduling ${MAINTENANCE_WINDOW_EVENTS_TASK_ID}, received ${e.message}`);
+  }
+}
+
+function registerMaintenanceWindowEventsGeneratorTask(
+  logger: Logger,
+  taskManager: TaskManagerSetupContract,
+  coreStartServices: Promise<[CoreStart, AlertingPluginsStart, unknown]>
+) {
+  taskManager.registerTaskDefinitions({
+    [MAINTENANCE_WINDOW_EVENTS_TASK_TYPE]: {
+      title: 'Maintenance window events generator task',
+      stateSchemaByVersion,
+      createTaskRunner: createEventsGeneratorTaskRunner(logger, coreStartServices),
+    },
+  });
+}
+
+export function createEventsGeneratorTaskRunner(
+  logger: Logger,
+  coreStartServices: Promise<[CoreStart, AlertingPluginsStart, unknown]>
+) {
+  return ({ taskInstance }: RunContext) => {
+    const state = taskInstance.state as LatestTaskStateSchema;
+    return {
+      async run() {
+        let totalMaintenanceWindowsWithGeneratedEvents = 0;
+        try {
+          const [{ savedObjects }] = await coreStartServices;
+
+          const savedObjectsClient = savedObjects.createInternalRepository([
+            MAINTENANCE_WINDOW_SAVED_OBJECT_TYPE,
+          ]);
+
+          const startRangeDate = moment().startOf('day').utc().toISOString();
```

Comment:
What will happen if the events of an MW expire just before the `startRangeDate`? Is it possible? I am wondering if we should start the day a couple of days before, just in case. Wdyt?